### DISCUSSION TITLE
Hide shaping points

### DISCUSSION THREAD

**Discussion by ramondevic** [devemux86/cruiser#427] [Q&A] [answered]
In a route I have many waypoints and many shape points.
I liked being able to easily and quickly hide the shape points and show only the waypoints.
On the navigation screen, when clicking on "recalculate route", on the "Routing > milestones" screen, ...
Maybe it could be added to the "Routing > Options" screen?

![Screenshot_20240123-184645](https://github.com/devemux86/cruiser/assets/122691401/831c11ea-cd52-41fc-944e-9ee4f6dc88cc)![Screenshot_20240123-184830](https://github.com/devemux86/cruiser/assets/122691401/8dbf4a11-bb7c-48af-b8b5-24a48a63139e)
![Screenshot_20240123-184852](https://github.com/devemux86/cruiser/assets/122691401/afb17978-3708-4af5-81d6-b5da4ab0fcc8)![Screenshot_20240123-184916](https://github.com/devemux86/cruiser/assets/122691401/1aaf1386-6b33-4e8e-8ced-b540d0bed311)

**Comment by devemux86** [comment]
@ramondevic 

There is an option in "Settings | Routing | Appearance" to select the zoom level
from which the shaping points are displayed on the map. Have you changed it?

**Reply by ramondevic** [reply]
Yes, I had already tried it but I thought it might be good if they could be hidden/shown in an easier and faster way, without having to make so many clicks, to see the development of the route and to be able to move them somewhere and return - to hide it in order not to disturb the vision of the route.

**Reply by devemux86** [reply]
@ramondevic do you mean show / hide them on the map?

(because you also show an image with rerouting in navigation)

We could think of an additional way to open the routing / appearance settings.
e.g. in the long press menu of the route.

**Reply by ramondevic** [reply]
On the map and in the others list I have put above with a mix between waypoints and shape points. I like see the waypoints but not the shapes (or yes), on map and the lists.

**Reply by devemux86** [reply]
@ramondevic each screen is different and we should discuss them with details.

For example the rerouting screen displays all waypoints: via + shaping points.
If we want to do rerouting with only via points, then we may need an extra setting.

**Reply by ramondevic** [reply]
When I prepare a long route, I do it in kurviger.de with all the waypoints I want to pass through (that's why I name them) and I modify the route by putting shape points to force the route to pass through this specific place but when the route is done I'm only interested in seeing the waypoints on the map and in all the lists, as you can see in the images in the first message of the thread.

But I think it should be a quick way to be able to choose whether or not I want to see the shape points in order to modify the route if needed.

Yes, an extra field "see shape points (yes/no)" would probably be needed

**Reply by devemux86** [reply]
@ramondevic

For the map there is already an option for shaping points in "Settings | Routing | Appearance".
(all route points have a similar option)

For the lists, which lists do you mean?
- Menu "Routing | Waypoints": how can we reorder waypoints if the shaping points are not in the list?

- Rerouting: we could add a new setting to do rerouting in navigation to the next via or shaping point?

**Reply by ramondevic** [reply]
In "Routes > Options" you could set "see shape points (yes/no)" and in the lists "Routes > waypoints", "Recalculate route", show both waypoints and shape points or only waypoints and be able to avoid seeing the data (shape points) that may not be of interest

![imatge](https://github.com/devemux86/cruiser/assets/122691401/b2dbe39c-bc52-4093-a90c-e6c1fd528162)
![299066786-afb17978-3708-4af5-81d6-b5da4ab0fcc8](https://github.com/devemux86/cruiser/assets/122691401/638f2f79-3244-4551-bb7f-0245af0212c1)![299066748-8dbf4a11-bb7c-48af-b8b5-24a48a63139e](https://github.com/devemux86/cruiser/assets/122691401/26a921ac-3b7c-4a1f-b261-9855f3ed0fae)

**Reply by devemux86** [reply, marked as answer]
@ramondevic

The first case for the appearance of shaping points on the map already exists in "Settings | Routing | Appearance".
(please see the image)

The other two cases (waypoints list and rerouting) can be discussed how they could be implemented.

<img src="https://github.com/devemux86/cruiser/assets/3484020/169f32be-30d0-4052-8b1f-cbc98fdbf189" width="50%"/>

**Reply by WolfgangZ42** [reply]
Hiding shaping points is perfectly solved on the map - in planning and in navigation! (I use 8 and 12)

In rerouting or changing the waypoint list (another form of rerouting) I can't really not imaging to hide the shaping points. I even insert a lot of shaping points on an existing route to minimize the difference of BRouter and Kurviger routing.

**Reply by ramondevic** [reply]
Yes, I know that you can hide or show shape points on the map depending on the zoom level you choose.

Regarding the editing/modification of the route it would be possible to show/hide the shape points in a quick way to be able to see the list in the edition and hide in order to see the final result.

Once on the road the shape points are more of a nuisance than a service.

**Reply by devemux86** [reply]
For rerouting we could add an option to use all points (via + shaping) or only via points?
Then the rerouting list will display only the points we want to use.

We have a similar setting in "Strict route following" (see image).

About the waypoints list, indeed this is more complicated how to handle it.

<img src="https://github.com/devemux86/cruiser/assets/3484020/16e4db48-86ee-4373-8380-92192a9763bf" width="50%"/>

**Reply by devemux86** [reply]
> I modify the route by putting shape points to force the route to pass through this specific place

> For rerouting we could add an option to use all points (via + shaping) or only via points?

Although we can all understand that these two things cannot easily agree with each other?

If we put shaping points in the route to shape it as we want and in rerouting display / use only the much less via points,
then a rerouting to next via point will probably create a more changed route, than if it was done to next shaping point.

**Reply by ramondevic** [reply]
No, rerouting already works well with via + shaping, it's just about having cleaner via + shaping data pages where you can choose whether to see via + shaping or just via.

**Reply by devemux86** [reply]
So if rerouting should work with all via + shaping points, then it should display all points so users can select them.

Then the suggestion remains for the waypoints list in routing menu?

**Reply by ramondevic** [reply]
All points (via + shape) should be displayed or only the waypoints depending on whether the user wants to modify/move points or only see the points (via) that interest him.

**Reply by devemux86** [reply]
@ramondevic 

Where do you mean exactly? In waypoints list?
Items cannot be reordered (drag up / down) in a list if the list does not display all items. Then the list will not be in edit mode.

**Reply by ramondevic** [reply]
Whether the list of waypoints can be shown (or not) the shapepoints.
That the user can hide/show the shapes as needed

You should more or less be able to see that if you want to edit/modify
![imatge](https://github.com/devemux86/cruiser/assets/122691401/41a13fa3-5c57-4b0d-ab8f-e7e621a2bbe9)


Or that if you are not interested in seeing everything
![imatge](https://github.com/devemux86/cruiser/assets/122691401/7129da8a-f33a-4a59-b9fd-86812df9e19b)

**Reply by WolfgangZ42** [reply]
> You should more or less be able to see that if you want to edit/modify

But - as Emux said:

> Items cannot be reordered (drag up / down) in a list if the list does not display all items. Then the list will not be in edit mode.

So this "shorten list" is only a display - nothing more ...

**Reply by ramondevic** [reply]
Yes, just for see places you have interes for stop, visiting, etc

**Comment by devemux86** [comment]
We could offer a simplified waypoints screen - _for viewing_ - with only the via points and their name / distance / time.

And as @MotoUKRider suggested [here](https://github.com/devemux86/cruiser/discussions/416#discussioncomment-8235991), it could also be offered during navigation for information purposes.
